src/main/java/com/capitalone/dashboard/client/RestClient.java

```diff
@@ -33,107 +35,107 @@ public RestClient(Supplier<RestOperations> restOperationsSupplier) {
 
     public ResponseEntity<String> makeRestCallPost(String url, String headerKey, String token, JSONObject body) {
         if (restOperations == null) { return null; }
 
+        HttpHeaders headers;
         if (StringUtils.isEmpty(headerKey) || StringUtils.isEmpty(token)) {
-            return makeRestCallPost(url, body);
+            headers = null;
+        } else {
+            headers = createHeaders(headerKey, token);
         }
 
-        long start = System.currentTimeMillis();
-
-        ResponseEntity<String> response
-                = restOperations.exchange(url, HttpMethod.POST, new HttpEntity<Object>(body, createHeaders(headerKey, token)), String.class);
-
-        long end = System.currentTimeMillis();
-
-        LOG.info("Time taken to make a post call to "+url+" = "+(end-start));
-
-        return response;
+        return this.makeRestCallPost(url, headers, body);
     }
 
     public ResponseEntity<String> makeRestCallPost(String url, RestUserInfo userInfo, JSONObject body) {
         if (restOperations == null) { return null; }
 
+        HttpHeaders headers;
         if ((userInfo == null)) {
-            return makeRestCallPost(url, body);
+            headers = null;
+        } else {
+            headers = createHeaders(userInfo.getFormattedString());
         }
 
-        long start = System.currentTimeMillis();
-
-        ResponseEntity<String> response
-                = restOperations.exchange(url, HttpMethod.POST, new HttpEntity<Object>(body, createHeaders(userInfo.getFormattedString())), String.class);
-
-        long end = System.currentTimeMillis();
+        return this.makeRestCallPost(url, headers, body);
+    }
 
-        LOG.info("Time taken to make a post call to "+url+" = "+(end-start));
+    private ResponseEntity<String> makeRestCallGet(String url, HttpEntity entity) throws RestClientException {
 
+        long start = System.currentTimeMillis();
+        ResponseEntity<String> response;
+        HttpStatus status = null;
+        try {
+            response = restOperations.exchange(url, HttpMethod.GET, entity, String.class);
+            status = response.getStatusCode();
+        } catch (HttpStatusCodeException e) {
+            status = e.getStatusCode();
+            throw e;
+        } finally {
+            long end = System.currentTimeMillis();
+            LOG.info("makeRestCall op=GET url=" + url + " status=" + status + " duration=" + (end - start));
+        }
         return response;
     }
 
     public ResponseEntity<String> makeRestCallGet(String url) throws RestClientException {
         if (restOperations == null) { return null; }
 
-        long start = System.currentTimeMillis();
-
-        ResponseEntity<String> response = restOperations.exchange(url, HttpMethod.GET, null, String.class);
-
-        long end = System.currentTimeMillis();
-
-        LOG.info("Time taken to make a get call to "+url+" = "+(end-start));
-
-        return response;
+        HttpEntity entity = null;
```

Comment:
why are we declaring `HttpEntity` if nothing is being done on it

Reply:
Same as above. Cannot pass directly null, must be cast. Changed to `return this.makeRestCallGet(url, (HttpEntity)null);` as you have a strong opinion on it.